src/faldbt/project.py

```diff
@@ -74,27 +71,30 @@ def __post_init__(self):
                 logger.debug(f"Non-generic test was not processed: {node.name}")
 
 
-@dataclass
 class DbtSource(_DbtNode):
-    name: str = field()
-    table_name: str = field()
-    unique_id: str = field()
-    tests: List[DbtTest] = field(init=False)
+    """Holds dbt source information."""
+
+    def __init__(
+        self,
+        name: str,
+        table_name: str,
+        unique_id: str,
+    ):
```

Comment:
In `DbtTest` we are receiving a `node` arg and then getting all the information needed from it. Should we have the same approach in both classes and receive a `node` here?

Maybe we should consider the same for `_DbtNode`

Reply:
I think that would be a nice solution as well (if it is viable). If we receive the `node` in all `_DbtNode` subclasses, then we can safely use a dataclass in the `_DbtNode` that takes a single argument (and extracts name etc.) and in each subclass we can do the additional computations in the `__post_init__` (and take additional arguments now, since they wont conflict). This way we don't loose any benefits of the dataclasses (e.g., the auto-generated repr & auto-completion support for various editors) and at the same time keep the code tidier. Though not sure there are any other things that might block this.

Reply:
I like @isidentical's suggestion. See updated code.